Modules/posixmodule.c

```diff
@@ -5490,12 +5490,43 @@ os__path_normpath_impl(PyObject *module, PyObject *path)
         return NULL;
     }
     Py_ssize_t norm_len;
-    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, &norm_len);
+    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, 0, &norm_len);
     PyObject *result = PyUnicode_FromWideChar(norm_path, norm_len);
     PyMem_Free(buffer);
     return result;
 }
 
+/*[clinic input]
+os._path_abspath
+
+    path: object
+
+    start: Py_ssize_t=0
```

Comment:
If this is meant to be a private method only, it should use positional-only parameters to ensure that the `METH_FASTCALL` calling convention is used.

1. Add this suggestion:
   ```suggestion
   os._path_abspath
   
       path: object
       start: Py_ssize_t = 0
       /
   ```
2. Regenerate clinic: `python3 Tools/clinic/clinic.py -f Modules/posixmodule.c`
3. `git commit`

Reply:
FTR, now the `METH_FASTCALL | METH_KEYWORDS` calling convention is used. I didn't measure the performance impact between those calling conventions. Perhaps it is negligible.